LSP-rust-analyzer.sublime-commands

```diff
@@ -15,8 +15,36 @@
         "caption": "LSP-rust-analyzer: Reload Project",
         "command": "rust_analyzer_reload_project"
     },
+    {
+        "caption": "LSP-rust-analyzer: Memory Usage (Clears Database)",
+        "command": "rust_analyzer_memory_usage"
+    },
+    {
+        "caption": "LSP-rust-analyzer: Run...",
+        "command": "rust_analyzer_run_project"
+    },
     {
         "caption": "LSP-rust-analyzer: Expand Macro Recursively",
         "command": "rust_analyzer_expand_macro"
+    },
+    {
+        "caption": "LSP-rust-analyzer: View Item Tree",
+        "command": "rust_analyzer_view_item_tree"
+    },
+    {
+        "caption": "LSP-rust-analyzer: View Syntax Tree",
+        "command": "rust_analyzer_syntax_tree"
+    },
+    {
+        "caption": "LSP-rust-analyzer: Open Cargo.toml",
+        "command": "rust_analyzer_open_cargo_toml"
+    },
+    {
+        "caption": "LSP-rust-analyzer: Find Matching Brace",
+        "command": "rust_analyzer_matching_brace"
+    },
+    {
+        "caption": "LSP-rust-analyzer: Join Lines",
+        "command": "rust_analyzer_join_lines"
```

Comment:
For something like that it would probably make sense to create a keybinding that overrides the default keybinding only if the session is active in the view. LSP keybindings have example of such cases.

Reply:
Would you have an example

Reply:
My memory has failed me. I was thinking of something like https://github.com/sublimelsp/LSP/blob/93dc0d9bc11969555131d45c0fddebb7387d4661/Default.sublime-keymap#L138-L150 but that checks for specific capability rather than the session name. I don't think we can currently do it.

Reply:
You could override `on_query_context` to provide something similar.

Reply:
Consider extending `on_query_context` in LSP to support that.

Reply:
Fixed in https://github.com/sublimelsp/LSP/pull/1763

Reply:
And released.

Reply:
I've removed join lines for now because there's no forward progress on this PR otherwise.